spaceapi/src/utils.rs

```diff
@@ -27,6 +80,7 @@ impl<T> Optional<T> {
         }
     }
 
+    /// Converts from `Optional<T>` to `Optional<&T>`
     pub fn as_ref<'r>(&'r self) -> Optional<&'r T> {
         match *self {
             Optional::Value(ref x) => Optional::Value(x),
```

Comment:
Could we also write `Optional::Value(x) => Optional::Value(&x)` instead? I find that a bit more readable.